### PR TITLE
Mhd detonation

### PR DESCRIPTION
**Pull request by Ishita-S**
## PR summary

This is the MHD Detonation problem. 

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
likewise, you can remove all of the SDC inputs/probins

File: Exec/hydro_tests/Sedov/GNUmakefile [hunk 1]
```diff
@@ -3,11 +3,11 @@ PROFILE    = FALSE
 
 DEBUG      = FALSE
 
-DIM        = 3
+DIM        = 2
```
Comment (zingale) [inline Exec/hydro_tests/Sedov/GNUmakefile:7]: can you revert these changes?

File: Exec/hydro_tests/Sedov/GNUmakefile [hunk 2]
```diff
@@ -3,11 +3,11 @@ PROFILE    = FALSE
...
 
 COMP	   = gnu
 
-USE_MPI    = TRUE
+USE_MPI    = FALSE
 USE_OMP    = FALSE
```
Comment (zingale) [inline Exec/hydro_tests/Sedov/GNUmakefile:11]: this one too

File: Exec/mhd_tests/Detonation/Problem.H [hunk 3]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (zingale) [inline Exec/mhd_tests/Detonation/Problem.H:1]: I don't think we need this file -- this was for the collisions stuff

File: Exec/mhd_tests/Detonation/Problem.f90 [hunk 4]
```diff
@@ -0,0 +1,91 @@
+! problem-specific Fortran stuff goes here
```
Comment (zingale) [inline Exec/mhd_tests/Detonation/Problem.f90:1]: I don't think we need this file either

File: Exec/mhd_tests/Detonation/Problem_F.H [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#include <AMReX_BLFort.H>
+
```
Comment (zingale) [inline Exec/mhd_tests/Detonation/Problem_F.H:2]: this also can go

File: Exec/mhd_tests/Detonation/bc_fill_nd.F90 [hunk 6]
```diff
@@ -0,0 +1,125 @@
+module bc_fill_module
```
Comment (zingale) [inline Exec/mhd_tests/Detonation/bc_fill_nd.F90:1]: I think we shouldn't do special boundary conditions, because I don't think these handle the magnetic fields.  So let's just use the normal BCs to start.

File: Exec/mhd_tests/Detonation/ext_src_nd.F90 [hunk 7]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (zingale) [inline Exec/mhd_tests/Detonation/ext_src_nd.F90:1]: this is just a stub and can be removed

**Comment by zingale** [comment]
I am not sure why the submodules are out of sync, maybe a git pull is needed there (for the `external/*`)?

**Comment by harpolea** [comment]
I think this may be because there was some conflict and we had to explicitly checkout the development branches of the submodules for things to work?

**Comment by zingale** [comment]
okay, once we tag Castro 20.09, I'll be updating the submodules, so then maybe this will just sort itself out.  Just waiting on AMReX.

**Comment by zingale** [comment]
we're switching to a new branch
